cmd/juju/model/setcredential.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for infos.
+
+package model
+
+import (
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/api/base"
+	cloudapi "github.com/juju/juju/api/cloud"
+	"github.com/juju/juju/api/modelmanager"
+	"github.com/juju/juju/cloud"
+	"github.com/juju/juju/cmd/juju/common"
+	"github.com/juju/juju/cmd/modelcmd"
+)
+
+// ModelCredentialAPI defines methods used to replace model credential.
+type ModelCredentialAPI interface {
+	Close() error
+	ChangeModelCredential(model names.ModelTag, credential names.CloudCredentialTag) error
+}
+
+// CloudAPI defines methods used to detemine if cloud credential exists on the controller.
+type CloudAPI interface {
+	Close() error
+	UserCredentials(names.UserTag, names.CloudTag) ([]names.CloudCredentialTag, error)
+	AddCredential(tag string, credential cloud.Credential) error
+}
+
+// modelCredentialCommand allows to change, replace a cloud credential for a model.
+type modelCredentialCommand struct {
+	modelcmd.ModelCommandBase
+
+	cloud      string
+	credential string
+
+	newAPIRootFunc            func() (base.APICallCloser, error)
+	newModelCredentialAPIFunc func(base.APICallCloser) ModelCredentialAPI
+	newCloudAPIFunc           func(base.APICallCloser) CloudAPI
+}
+
+func NewModelCredentialCommand() cmd.Command {
+	command := &modelCredentialCommand{
+		newModelCredentialAPIFunc: func(root base.APICallCloser) ModelCredentialAPI {
+			return modelmanager.NewClient(root)
+		},
+		newCloudAPIFunc: func(root base.APICallCloser) CloudAPI {
+			return cloudapi.NewClient(root)
+		},
+	}
+	command.newAPIRootFunc = func() (base.APICallCloser, error) {
+		return command.NewControllerAPIRoot()
+	}
+	return modelcmd.Wrap(command)
+}
+
+// Info implements Command.Info.
+func (c *modelCredentialCommand) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:    "set-credential",
+		Args:    "<cloud name> <credential name>",
+		Purpose: "Change cloud credential for the current or specified model.",
+		Doc:     modelCredentialDoc,
+	}
+}
+
+// SetFlags implements Command.SetFlags.
+func (c *modelCredentialCommand) SetFlags(f *gnuflag.FlagSet) {
```

Comment:
You don't need to define this if you're just delegating to ModelCommandBase.

Reply:
true, thanks :D